### ISSUE TITLE
[Feature Suggestion] Add histogram plot of residual errors to ResidualsPlot

### ISSUE THREAD

**Issue by ianozsvald** [DistrictDataLabs/yellowbrick#264]
The current `ResidualsPlot` shows training and testing residuals as a scatter plot, by eye we can get an idea of whether more errors are above or below the 0 line. By adding a histogram of testing errors we might more clearly be able to tell if errors have a Normal  distribution.

In the following examples I have some large positive and negative errors, from the histogram it looks as though I have a negatively skewed distribution which might tell me something about my training examples:
![image](https://user-images.githubusercontent.com/273210/27382522-40ac05a4-567f-11e7-927b-7c724a0faa0f.png)

```
from yellowbrick.regressor import ResidualsPlot
fig, ax = plt.subplots(figsize=(8,6)); 
model = ResidualsPlot(clone_estimator(clf), ax=ax)
model.fit(X_train, y_train)
model.score(X_test, y_test)

# add histogram of residual errors
left, bottom, width, height = [0.65, 0.17, 0.2, 0.2]
ax2 = fig.add_axes([left, bottom, width, height])

testing_residuals = pd.Series(model.predict(X_test) - y_test)
testing_residuals.plot(kind="hist", bins=50, title="Residuals on Predicted", ax=ax2);
ax2.vlines(0, ymin=0, ymax=ax2.get_ylim()[1] ) # add x==0 line

model.poof()
```

It isn't obvious where the best location would be for the histogram. Annoyingly I cannot get an `alpha` value for `ax2` either (I'd hoped to make this semi-transparent so location was less of an issue).

**Comment by ndanielsen** [comment]
@ianozsvald thanks for this great feature enhancement suggestion. you're welcome to make a pull request with this enhancement =)

**Comment by ndanielsen** [comment]
@bbengfort

**Comment by ianozsvald** [comment]
Sorry, I'm only going as far as sharing some proof-of-concept code, I'm still recovering from running PyDataLondon 2017 and I'm not up to extending any libraries at the moment! Maybe someone will find time to take this a little further. Cheers!

**Comment by ndanielsen** [comment]
@ianozsvald thank you so much for your feature suggestions and the time that you've spent in writing up the detailed issues with examples. it is really fantastic and helpful. i'm sure that someone will move forward with these.

**Comment by ianozsvald** [comment]
Well, thank you all too for putting this library together, I've got a bunch of my own hacky viz tools but you've built something far more useful here. @rebeccabilbro's talk for us at the conference (and the book signing she joined me for) was ace :-)

**Comment by rebeccabilbro** [comment]
Hey there @ianozsvald - thanks so much for all your work on PyData London - what a terrific conference! And thanks for checking out Yellowbrick -- I love this idea of plotting error distributions to make it easier to look for things like skew and heavy tails. As for how to best cope with subaxis locations, we might take a look at [GridSpec](https://matplotlib.org/api/gridspec_api.html#matplotlib.gridspec.GridSpec), similar to what @pdamodaran did with [JointPlot](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/features/jointplot.py#L208).

**Comment by bbengfort** [comment]
@pdamodaran @ianozsvald  my instinct actually says to put it on the right side, oriented vertically so that the histogram shares an access with the residuals (the y axis of the plot). In this orientation I think it might be easier to directly compare and would probably also have the effect of balancing the axes so that zero is in the middle, RE: the axes issues we had in #263 -- what do you guys think?

**Comment by bbengfort** [comment]
@ianozsvald sorry for the delay on this; we've been slammed - but I wanted to take a crack at this when you reminded me via the notebooks in the Slack channel the other day. 

What do you think about this?

![residuals_hist_annotation](https://user-images.githubusercontent.com/745966/41488599-dd7b7f6c-70ba-11e8-925f-f59bcfb2a2df.png)

It's an initial prototype, but I'm thinking we'll just make it an option right now (e.g. `hist=True`) and then this is displayed.

**Comment by ianozsvald** [comment]
Brilliant! This feels more informative than mine. I defo prefer your orientation for the histogram and the larger size is clearly better. I wonder if a normalised histogram makes more sense (as the test set is likely to be smaller than the training set, which won't aid comparison)?

**Comment by bbengfort** [comment]
@ianozsvald I'm happy to add a normalization argument here - so instead of a frequency, use a PDF and stack training and test rather than overlay test on top of training?

**Comment by ianozsvald** [comment]
Overlaying test and train probably looks fine (given the PDF), that should be more comparable than having one stacked on the other? The PDF should look lovely regardless. I look forward to seeing it :-)
